### PR TITLE
Refactor some code

### PR DESCRIPTION
**Pull request by ZenoTan**
solve #20

### CODE REVIEW

File: bench.go [hunk 1]
```diff
@@ -82,43 +79,29 @@ func (s *scaleOut) Run() error {
...
+	// if low deviation in a series of scores applies to all stores, then it is balanced.
+	for _, store := range matrix {
+		if len(store) != 10 {
 			return false, nil
 		}
 		mean := 0.0
 		dev := 0.0
-		for _, v := range data.Values {
-			mean += float64(v.Value) / 10
+		for _, score := range store {
```
Comment (lhy1024) [inline bench.go]: stores?

File: cluster.go [hunk 2]
```diff
@@ -138,3 +145,62 @@ func (c *Cluster) GetLastReport() (*WorkloadReport, error) {
 	}
 	return &reports[0], nil
 }
+
+func (c *Cluster) getMetric(query string, t time.Time) (float64, error) {
```
Comment (lhy1024) [inline cluster.go]: do we need some tests for them?

File: bench.go [hunk 3]
```diff
@@ -157,124 +140,59 @@ func (s *scaleOut) Collect() error {
...
-	}
-	if len(warnings) > 0 {
-		log.Warn("query has warnings")
-	}
-	vector = result.(model.Vector)
-	if len(vector) >= 1 {
-		rep.CurBalanceLeaderCount = int(vector[0].Value)
+		return "", err
 	}
+	rep.CurBalanceLeaderCount = int(value)
```
Comment (lhy1024) [inline bench.go]: I think pre-cur with these code can be changed into a function

File: cluster.go (hunk 2 above)
Comment (ZenoTan) [reply cluster.go]: We can mock it in the future.

Comment (lhy1024) [reply cluster.go]: I think it may be difficult to mock promethus

Comment (ZenoTan) [reply cluster.go]: For br and prometheus, it seems like we need a stable instance to test it...

**Review by lhy1024 — COMMENTED** [review]
LGTM, we need manual test

**Comment by ZenoTan** [comment]
Already finished a manual test
image

**Comment by lhy1024** [comment]
Good, plz fix conflict

**Comment by ZenoTan** [comment]
Still to resolve conflicts

File: bench/bench.go [hunk 4]
```diff
@@ -157,9 +157,6 @@ func (s *scaleOut) Collect() error {
 
 func (s *scaleOut) queryPrevCur(query string, prevArg, curArg interface{}, typ int) error {
 	prevValue, err := s.c.getMetric(query, s.t.addTime)
-	if err != nil {
-		return err
-	}
```
Comment (lhy1024) [inline bench/bench.go]: we need to keep it

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**